dbt-bigquery/src/dbt/include/bigquery/macros/materializations/incremental_strategy/insert_overwrite.sql

```diff
@@ -40,62 +40,144 @@
+  -- explicitly listed it in `partitions=`.
+#}
+{% macro bq_static_copy_partitions_insert_overwrite_sql(
+  tmp_relation, target_relation, sql, partition_by, partitions, tmp_relation_exists
 ) %}
 
-      {% set predicate -%}
-          {{ partition_by.render_wrapped(alias='DBT_INTERNAL_DEST') }} in (
-              {{ partitions | join (', ') }}
-          )
-      {%- endset %}
+    {%- if tmp_relation_exists is false -%}
+        {%- set source_sql = bq_static_select_insert_overwrite_sql(tmp_relation, sql, partition_by, tmp_relation_exists) %}
 
-      {%- set source_sql -%}
-        (
-          {% if partition_by.time_ingestion_partitioning and tmp_relation_exists -%}
-          select
-            {{ partition_by.insertable_time_partitioning_field() }},
-            * from {{ tmp_relation }}
-          {% elif tmp_relation_exists -%}
-            select
-            * from {{ tmp_relation }}
-          {%- elif partition_by.time_ingestion_partitioning -%}
-            {{ wrap_with_time_ingestion_partitioning_sql(partition_by, sql, True) }}
-          {%- else -%}
-            {{sql}}
-          {%- endif %}
-
-        )
-      {%- endset -%}
+        {# -- we run temp table creation in a separate script to move to partitions copy if it doesn't already exist #}
+        {%- call statement('create_tmp_relation_for_copy', language='sql') -%}
+          {{ bq_create_table_as(partition_by, true, tmp_relation, source_sql, 'sql')
+        }}
+        {%- endcall %}
+    {%- endif -%}
 
-      {% if copy_partitions %}
-          {% do bq_copy_partitions(tmp_relation, target_relation, partitions, partition_by) %}
-      {% else %}
+    {%- set partitions_sql -%}
+	select
+	    cast(partition_literal as timestamp) as partition_ts
+	from unnest([
+	    {{ partitions | join(', ') }}
+	]) as partition_literal
+    {%- endset -%}
 
-      {#-- In case we're putting the model SQL _directly_ into the MERGE statement,
-         we need to prepend the MERGE statement with the user-configured sql_header,
-         which may be needed to resolve that model SQL (e.g. referencing a variable or UDF in the header)
-         in the "temporary table exists" case, we save the model SQL result as a temp table first, wherein the
-         sql_header is included by the create_table_as macro.
-      #}
-      -- 1. run the merge statement
-      {{ get_insert_overwrite_merge_sql(target_relation, source_sql, dest_columns, [predicate], include_sql_header = not tmp_relation_exists) }};
+    {%- set resolved_partitions = run_query(partitions_sql).columns[0].values() -%}
 
-      {%- if tmp_relation_exists -%}
-      -- 2. clean up the temp table
-      drop table if exists {{ tmp_relation }};
-      {%- endif -%}
+    {% do bq_copy_partitions(tmp_relation, target_relation, resolved_partitions, partition_by) %}
 
-  {% endif %}
+    {# clean up temp table #}
+    drop table if exists {{ tmp_relation }}
```

Comment:
nit: missing semi-colon
```suggestion
    drop table if exists {{ tmp_relation }};
```

Reply:
Is it going to work? I remember that I had issues in the past when I put semi-colon at the end of a statement (I felt like dbt was adding one and it failed because there was 2 semi-colons).